client/src/ui/ClassViewNew.tsx

```diff
@@ -0,0 +1,116 @@
+
+  useEffect(() => {
+    async function updateCurrentClass(number: number, subject: string) {
+      const response = await axios.post(`/v2/getCourseByInfo`, {
+        number,
+        subject: subject.toLowerCase(), // TODO: fix backend to handle this
+      });
+
+      const course = response.data.result;
+      if (course) {
+        setSelectedClass(course);
+        setPageStatus(PageStatus.Success);
+      } else {
+        setPageStatus(PageStatus.Error);
+      }
+    }
+    updateCurrentClass(number, subject);
+  }, [number, subject]);
+
+  if (pageStatus === PageStatus.Error) {
+    return (
+      <div className={`row ${styles.errorContainer}`}>
+        {/* TODO: no props on orig implementation */}
+        <Navbar userInput={input} />
+        <img
+          className={styles.errorGauge}
+          src="/error.svg"
+          width="100vw"
+          height="auto"
+          alt="error"
+        />
+        <h2>Sorry, we couldn't find the class you're searching for.</h2>
+        <h2>Please search for a different class.</h2>
+      </div>
+    );
+  }
+
+  if (pageStatus === PageStatus.Success && !!selectedClass) {
+    courseVisited(selectedClass?.classSub, selectedClass?.classNum);
+    return (
+      <div className={`row ${styles.classView}`}>
+        <Navbar userInput={input} />
+        <div className={`${styles.content}`}>
+          <div className={`col-lg-4 col-md-5 ${styles.courseInfoColumn}`}>
+            <div>
+              <h1 className={styles.courseTitle}>{selectedClass.classTitle}</h1>
+              <p className={styles.courseSubtitle}>
+                {selectedClass.classSub.toUpperCase() +
+                  " " +
+                  selectedClass.classNum +
+                  ", " +
+                  lastOfferedSems(selectedClass)}
+              </p>
+            </div>
+            {/* TODO: show button for leaving a review on sm/xs screens */}
+            <div
+              className={`hidden-sm hidden-xs ${styles.reviewFormContainer}`}
+            >
+              <Form course={selectedClass} inUse={true} />
+            </div>
+          </div>
+          <div className={`col-lg-8 col-md-7 ${styles.courseReviewColumn}`}>
+            <div className={styles.gaugeContainer}>
+              <div className={styles.gauge}>
+                <Gauge rating={selectedClass!.classRating} label="Overall" />
+              </div>
+              <div className={styles.gauge}>
+                <Gauge
+                  rating={selectedClass.classDifficulty}
+                  label="Difficulty"
+                />
+              </div>
+              <div className={styles.gauge}>
+                <Gauge rating={selectedClass.classWorkload} label="Workload" />
+              </div>
+            </div>
+            <h2 className={styles.pastReviews}>Past Reviews</h2>
+            <div className={styles.courseReviews}>
+              <CourseReviews
```

Comment:
I'm seeing a double scroll bar in the past reviews section, not sure why that is happening.